torchao/quantization/quant_primitives.py

```diff
@@ -159,15 +159,15 @@ def _get_reduction_params(block_size, input_size):
             cur_dim += 1
     return shape_for_reduction, reduction_dims
 
-
+@torch.no_grad()
 def quantize_affine(
     input: torch.Tensor,
     block_size: Tuple[int, ...],
     scale: torch.Tensor,
     zero_point: Optional[torch.Tensor],
     output_dtype: torch.dtype,
-    quant_min: Optional[int] = None,
-    quant_max: Optional[int] = None,
+    quant_min: Optional[Union[int, float]] = None,
+    quant_max: Optional[Union[int, float]] = None,
     zero_point_domain: ZeroPointDomain = ZeroPointDomain.INT,
```

Comment:
I think you'll need to update this type to be Optional[ZeroPointDomain] as well right?

Reply:
Currently infer_schema in torch doesn't support Optional[Union[int, float]]

Reply:
you mean the export test will fail?

Reply:
can you point me to the test?